src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -1156,9 +1155,7 @@ internal static string ToStringEnumerable(ExecutionContext context, IEnumerable
                 return string.Empty;
             }
 
-            int separatorLength = separatorToUse.Length;
-            returnValue.Remove(returnValue.Length - separatorLength, separatorLength);
-            return returnValue.ToString();
+            return returnValue.ToString(0, returnValue.Length - separatorToUse.Length);
```

Comment:
You may want to change the similar code in `ToStringEnumerator` as well.

Reply:
Done.